core/src/main/scala/org/http4s/rho/bits/ResultMatcher.scala

```diff
@@ -345,103 +351,108 @@ object ResultMatcher {
       INSUFFICIENTSTORAGE,
       LOOPDETECTED,
       NOTEXTENDED,
-      NETWORKAUTHENTICATIONREQUIRED]): Task[Response] = Task.now(r.resp)
+      NETWORKAUTHENTICATIONREQUIRED])(implicit F: Monad[F]): F[Response[F]] = F.pure(r.resp)
```

Comment:
Why Monad over Applicative if we only use `pure`?

Reply:
Because this extends the `ResultMatcher` trait, and some of the other implementations use `flatMap`.